scripts/create-k3s-cluster-with-SSL-passthrough.sh

```diff
@@ -81,6 +81,29 @@ fi
 mkdir -p ~/.kube
 export KUBECONFIG=~/.kube/config
 sudo kubectl --kubeconfig=/etc/rancher/k3s/k3s.yaml config view --raw > "$KUBECONFIG"
+
+# Wait for k3s service to be ready and kubectl to work
+echo "Waiting for k3s service to be ready via socket file..."
+timeout=300
+elapsed=0
+while [ $elapsed -lt $timeout ]; do
+    if [ -S /run/k3s/containerd/containerd.sock ] && kubectl get nodes >/dev/null 2>&1; then
+        echo " K3s socket exists and kubectl is working! (${elapsed}s)"
+        break
+    fi
+    echo "⌛ K3s not ready yet, waiting... (${elapsed}s/${timeout}s)"
+    sleep 5
+    elapsed=$((elapsed + 5))
+done
+
+if [ $elapsed -ge $timeout ]; then
+    echo "❌ Timed out waiting for K3s to be ready."
+    exit 1
+fi
+
+echo "Waiting for k3s cluster nodes to be ready..."
+kubectl wait --for=condition=ready nodes --all --timeout=300s
```

Comment:
Why is this wait here? The following commands do not depend on the cluster's Node being "ready".